src/test/java/it/com/atlassian/bitbucket/jenkins/internal/trigger/TriggerWebhookCreationTest.java

```diff
@@ -0,0 +1,144 @@
+package it.com.atlassian.bitbucket.jenkins.internal.trigger;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketRepository;
+import com.atlassian.bitbucket.jenkins.internal.trigger.BitbucketWebhookMultibranchTrigger;
+import com.atlassian.bitbucket.jenkins.internal.trigger.BitbucketWebhookTriggerImpl;
+import com.atlassian.bitbucket.jenkins.internal.trigger.events.BitbucketWebhookEvent;
+import hudson.model.FreeStyleProject;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.BitbucketJenkinsRule;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.BitbucketTestClient;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.GitHelper;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.JenkinsProjectHandler;
+import jenkins.branch.MultiBranchProject;
+import jenkins.scm.api.SCMSource;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.Timeout;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.atlassian.bitbucket.jenkins.internal.trigger.events.BitbucketWebhookEvent.*;
+import static it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+public class TriggerWebhookCreationTest {
+
+    private static final Set<String> PR_EVENTS = new HashSet<>();
+    private static final Set<String> PR_EVENTS_PRE_7_6 = new HashSet<>();
+    private static String cloneUrl;
+    private static String projectKey;
+    private static String repoSlug;
+    @Rule
+    public final BitbucketJenkinsRule bbJenkinsRule = new BitbucketJenkinsRule();
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @Rule
+    public final Timeout testTimeout = new Timeout(0, TimeUnit.MINUTES);
+    private GitHelper gitHelper;
+    private JenkinsProjectHandler projectHandler;
+    private BitbucketTestClient testClient;
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        deleteRepository(projectKey, repoSlug);
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        String repoName = REPO_NAME + "-fork";
+        BitbucketRepository repository = forkRepository(PROJECT_KEY, REPO_SLUG, repoName);
+        repoSlug = repository.getSlug();
+        projectKey = repository.getProject().getKey();
+        cloneUrl =
+                repository.getCloneUrls()
+                        .stream()
+                        .filter(repo ->
+                                "http".equals(repo.getName()))
+                        .findFirst()
+                        .orElse(null)
+                        .getHref();
```

Comment:
```suggestion
                        .map(BitbucketNamedLink::getHref)
                        .orElse(null);
```
I know this is testing code ;) but this would avoid the NPE

Reply:
Would it not be better to get the NPE here, rather than later? Happy to change though